lib/datadog/core/workers/async.rb

```diff
@@ -10,6 +10,7 @@ module Thread
           FORK_POLICY_STOP = :stop
           FORK_POLICY_RESTART = :restart
           SHUTDOWN_TIMEOUT = 1
+          GLOBAL_MUTEX = Mutex.new
```

Comment:
```suggestion
          # This single shared mutex is used to avoid concurrency issues during the
          # per-instance lazy-initialized mutexes.
          MUTEX_INIT = Mutex.new
```